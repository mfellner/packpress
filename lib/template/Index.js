import React, { Component } from 'react'

export default class Index extends Component {
  props: {
    Root: React$Component;
    language: string;
    title: string;
    description: string;
  }

  render() {
    const {Root, language, title, description, ...other} = this.props
    return <html lang={language}>
      <head>
        <meta charSet="utf-8"/>
        <meta httpEquiv="x-ua-compatible" content="ie=edge"/>
        <title>{title}</title>
        <meta name="description" content={description}/>
        <meta name="viewport" content="width=device-width, initial-scale=1"/>
      </head>
      <body>
        <main id="main">
          <Root {...other}/>
        </main>
      </body>
    </html>
  }
}
