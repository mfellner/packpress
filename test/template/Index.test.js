import React from 'react'
import renderer from 'react-test-renderer'
import Index from '../../lib/template/Index'
import Test from '../fixtures/Test'

describe('Index', () => {
  it('should render the Root property component', () => {
    const props = {
      Root: Test,
      language: 'en',
      title: 'Test',
      description: 'This is only a test.',
      text: 'Hello, world!',
      link: 'https://github.com/mfellner/packpress'
    }
    const component = renderer.create(
      <Index {...props}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
