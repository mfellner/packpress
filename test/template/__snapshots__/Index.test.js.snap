exports[`Index should render the Root property component 1`] = `
<html
  lang="en">
  <head>
    <meta
      charSet="utf-8" />
    <meta
      content="ie=edge"
      httpEquiv="x-ua-compatible" />
    <title>
      Test
    </title>
    <meta
      content="This is only a test."
      name="description" />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport" />
  </head>
  <body>
    <main
      id="main">
      <span>
        <a
          href="https://github.com/mfellner/packpress">
          Hello, world!
        </a>
      </span>
    </main>
  </body>
</html>
`;
